base/number.jl

```diff
@@ -11,6 +11,8 @@ length(x::Number) = 1
 endof(x::Number) = 1
 getindex(x::Number) = x
 getindex(x::Number, i::Integer) = i == 1 ? x : throw(BoundsError())
+getindex(x::Number, i1::Integer, i2::Integer) = 1 == i1 == i2 ? x : throw(BoundsError())
+getindex(x::Number, i1::Integer, i2::Integer, i3::Integer) = 1 == i1 == i2 == i3 ? x : throw(BoundsError())
```

Comment:
The reasons of changes in unrelated files are:
* In sysimg.jl: include sparse.jl before broadcast.jl, needed because SparseMatrixCSC type is now used inside broadcast.jl.
* In cholmod.jl: after the changes in sysimg.jl julia build failed. The reason was that operator .== was being used in cholmod.jl, before it was defined in broadcast.jl. I repaced the operator by equivalent code. This change could probably be avoided by reorganizing the code in the modules (and not modying sysimg.jl), but I think it's cleaner having all broadcasting code together.
* In number.jl: this change makes broadcasting of the types <NUMBER> <OP> <ARRAY> and <ARRAY> <OP> <NUMBER> much faster for the common case of 2 and 3 dimensional arrays (for SparseMatrixCSC, only the 2 dimensions case is relevant).